### PR TITLE
Fix TagList formwidget

### LINKED ISSUE #1223: TagList not saving an empty value
It appears that a `TagList` widget in `relation` mode is not saving an empty value, because it no longer sends a POST value when empty.

### Replication steps

In the Test plugin:

- Go to **Users**, edit either user.
- Click **Tag List Field** in the top alternative views box.
- Empty the **Roles** field, then try and save.

It does not fail validation, so the form saves, as the last value in the tag list is kept on refresh even though all tags are removed.

### PR DESCRIPTION
**Pull request by mjauvin**
The TagList formwidget was not working properly in relation mode if the relation was not using a pivot table (e.g. morphMany/hasMany)

All modes did not update the model field or relation when all tags had been cleared.

Fixes #1223

### CODE REVIEW

**Comment by mjauvin** [comment]
@bennothommo I just saw you started a PR on TagList as well, thought you might want to review this one.

**Comment by mjauvin** [comment]
This PR fixes the TagList in relation mode for MorphMany relations, but looks like it breaks it for BelongsToMany relations.

**Comment by bennothommo** [comment]
@mjauvin it's probably worth adding an example in the Test plugin if there isn't one already, if adding the unit tests becomes too difficult.

**Comment by mjauvin** [comment]
yeah, that's what I'm going to do for now.

**Comment by mjauvin** [comment]
tested working with the following modes using Winter.Test plugin in Post & Record models:

- relation (belongsToMany, hasMany, morphMany)
- array (useKey = true/false)
- string (useKey = true/false)

All the above tests also include clearing all values and making sure the model field or relation was updated accordingly.

File: modules/backend/widgets/Form.php [hunk 1]
```diff
@@ -1227,7 +1227,11 @@ public function getSaveData(): array
             // Exclude fields that didn't provide any value
             $fieldValue = $this->dataArrayGet($result, $parts, FormField::NO_SAVE_DATA);
             if ($fieldValue === FormField::NO_SAVE_DATA) {
-                continue;
+                if ($widget->getConfig('type') === 'taglist') {
+                    $fieldValue = in_array($widget->getConfig('mode'), ['array','relation']) ? [] : '';
+                } else {
+                    continue;
+                }
```
Comment (mjauvin) [inline modules/backend/widgets/Form.php]: The above code snippet should not be required if we can fix the TagList widget to include empty array or empty string in the POST data.

@bennothommo any help would be appreciated.

Comment (mjauvin) [reply modules/backend/widgets/Form.php]: fixed.

**Comment by mjauvin** [comment]
@LukeTowers Anything more before merging this?

**Comment by LukeTowers** [comment]
@mjauvin @bennothommo Has it been fully tested?

**Comment by mjauvin** [comment]
> @mjauvin @bennothommo Has it been fully tested?

Yes, see previous comment above.

**Comment by AIC-BV** [comment]
Can confirm this fixes the problem
Thanks!

**Comment by AIC-BV** [comment]
image
Oh it seems to generate 2 tags instead of one when creating a tag for the first time

When removing all tags, they also save as an empty string
image

When adding tags
image

**Comment by mjauvin** [comment]
@AIC-BV can you show your tag field definition and relation?

**Comment by AIC-BV** [comment]
@mjauvin 
There is no relation

```
tags:
    label: Tags
    type: taglist
    customTags: false
    options:
        nieuw: Nieuw
        best: Beste deal
    span: auto
```

**Comment by LukeTowers** [comment]
@AIC-BV can you open an issue for that so that we don't release 1.2.8 until it's fixed?

**Comment by AIC-BV** [comment]
https://github.com/wintercms/winter/issues/1275

**Pull request merged**